src/Escrow.sol

```diff
@@ -249,32 +275,109 @@ contract Escrow {
+    /**
+     * @notice Deposits escrow without assigning a receiver.
+     * @param sender Address of the sender to be credited with the deposited amount.
+     * @param amount Amount of escrow to deposit.
+     * @dev The escrow must be approved for transfer by the caller.
+     * @dev REVERT: this function will revert if the escrow transfer fails.
+     */
+    function depositUnassigned(address sender, uint256 amount) external {
+        unassignedAccounts[sender] += amount;
+        escrowToken.safeTransferFrom(msg.sender, address(this), amount);
+        emit UnassignedDeposit(msg.sender, sender, amount);
+    }
+
+    /**
+     * @notice Assigns unassigned escrow to a receiver.
+     * @param receiver Address of the receiver.
+     * @param amount Amount of escrow to assign.
+     * @dev Unassigned escrow must be previously deposited via `depositUnassigned()`.
+     * @dev REVERT: this function will revert if the sender does not have enough unassigned escrow.
+     */
+    function assignDeposit(address receiver, uint256 amount) external {
+        address sender = msg.sender;
+        uint256 unassignedBalance = unassignedAccounts[sender];
+
+        // Check sender has enough unassigned balance
+        if (unassignedBalance < amount) {
+            revert InsufficientUnassignedEscrow({
+                available: unassignedBalance,
+                required: amount
+            });
+        }
+
+        // Assign deposit
+        unassignedAccounts[sender] -= amount;
+        escrowAccounts[sender][receiver].balance += amount;
+
+        emit DepositAssigned(sender, receiver, amount);
+    }
+
+    /**
+     * @notice Assigns unassigned escrow for multiple receivers.
+     * @param receivers Array of addresses of the receivers.
+     * @param amounts Array of amounts of escrow to deposit.
+     * @notice Unassigned escrow must be previously deposited via `depositUnassigned()`.
+     * @dev REVERT: this function will revert if the sender does not have enough unassigned escrow.
+     * @dev REVERT: if the length of the receivers and amounts arrays do not match.
+     */
+    function assignDepositMany(
+        address[] calldata receivers,
+        uint256[] calldata amounts
+    ) external {
+        if (receivers.length != amounts.length) {
+            revert InputsLengthMismatch();
+        }
+
+        address sender = msg.sender;
+        uint256 unassignedBalance = unassignedAccounts[sender];
+
+        // Assign deposits
+        uint256 totalAssigned = 0;
+        for (uint256 i = 0; i < receivers.length; i++) {
+            totalAssigned += amounts[i];
+            escrowAccounts[sender][receivers[i]].balance += amounts[i];
+            emit DepositAssigned(sender, receivers[i], amounts[i]);
+        }
+
+        // Check sender has enough unassigned balance
+        // NOTE: this check is done after the deposits are assigned to avoid looping 
+        //       twice. It's expected the sender will know it's unassigned balance.
```

Comment:
nit: "it's unassigned balance" -> "its unassigned balance" or "their unassigned balance"